### PR TITLE
v2

### PR DESCRIPTION
**Pull request by SubJunk**
more details are in the commit messages which i tried to keep fairly scoped

### CODE REVIEW

File: src/controllers/media.ts [hunk 1]
```diff
@@ -93,23 +90,105 @@ const getFromIMDbAPI = async(imdbId?: string, searchRequest?: SearchRequest): Pr
...
 };
 
+/**
+ * Attempts a query to the IMDb API and standardizes the response
+ * before returning.
+ *
+ * IMDb does not return goofs, osdbHash, tagline, trivia
+ *
+ * @param imdbId the IMDb ID
+ * @param searchRequest a query to perform in order to get the imdbId
```
Comment (js-kyle) [inline src/controllers/media.ts:100]: JSDoc is missing two parameters

File: src/controllers/media.ts [hunk 2]
```diff
@@ -93,23 +90,105 @@ const getFromIMDbAPI = async(imdbId?: string, searchRequest?: SearchRequest): Pr
...
+ */
+const getFromIMDbAPIV2 = async(imdbId?: string, searchRequest?: SearchRequest, seasonNumber?, episodeNumber?): Promise<MediaMetadataInterface> => {
+  if (!imdbId && !searchRequest) {
+    throw new Error('All parameters were falsy');
```
Comment (js-kyle) [inline src/controllers/media.ts:104]: Could be more accurate/descriptive

File: src/controllers/media.ts [hunk 3]
```diff
@@ -252,6 +331,91 @@ export const getBySanitizedTitle = async(ctx: Context): Promise<MediaMetadataInt
     await FailedLookups.updateOne(failedLookupQuery, { $inc: { count: 1 } }, { upsert: true, setDefaultsOnInsert: true }).exec();
     throw new MediaNotFoundError();
   }
+};
+
+/**
+ * Looks up a video by its title, and optionally its year, season and episode number.
+ * If it is an episode, it also sets the series data.
+ *
+ * @param ctx
```
Comment (js-kyle) [inline src/controllers/media.ts:340]:
```suggestion
 * @param ctx Koa request object
```

File: src/controllers/media.ts (hunk 1 above)
Comment (SubJunk) [reply src/controllers/media.ts]: Done

File: src/controllers/media.ts (hunk 2 above)
Comment (SubJunk) [reply src/controllers/media.ts]: Done

File: src/controllers/media.ts (hunk 3 above)
Comment (SubJunk) [reply src/controllers/media.ts]: Done

**Review by js-kyle — APPROVED** [review]
Keen to remove some of the duplication soon, can look at it in a different PR

**Pull request merged**